eslint-config/resolved-configs/generate.js

```diff
@@ -0,0 +1,331 @@
+          ...process.env,
+        },
+      },
+      (error, stdOut, stdErr) => {
+        if (error) {
+          reject(error);
+          return;
+        }
+
+        if (stdErr) {
+          reject(stdErr);
+          return;
+        }
+
+        if (stdOut.toString().trim() === 'undefined') {
+          resolve(undefined);
+        } else {
+          resolve(stdOut);
+        }
+      }
+    );
+  });
+
+  let resolved = resolvedConfigJSON
+    ? JSON.parse(resolvedConfigJSON)
+    : undefined;
+
+  if (resolved) {
+    resolved = postProcessConfig(resolved);
+  } else {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `\x1b[33m[WARNING] ESLint could not resolve config for file "${input.testInputFile}"\x1b[0m`
+    );
+  }
+
+  return {
+    ...input,
+    resolved,
+  };
+}
+
+const TAG_DELIMITER = '_';
+
+function getConfigVariations() {
+  let current = ['base'];
+
+  current = [
+    ...current,
+    ...current.map((x) => `${x}${TAG_DELIMITER}typescript`),
+  ];
+
+  current = [
+    ...current,
+    ...current.map((x) => `${x}${TAG_DELIMITER}react`),
+    ...current.map(
+      (x) => `${x}${TAG_DELIMITER}react${TAG_DELIMITER}react-redux`
+    ),
+  ];
+
+  current = [
+    ...current,
+    ...current.map((x) => `${x}${TAG_DELIMITER}jest`),
+    ...current.map((x) => `${x}${TAG_DELIMITER}vitest`),
+  ];
+
+  current = [
+    ...current,
+    ...current
+      .filter((x) => {
+        const tags = x.split(TAG_DELIMITER);
+        return (
+          tags.includes('react') &&
+          (tags.includes('jest') || tags.includes('vitest'))
+        );
+      })
+      .map((x) => `${x}${TAG_DELIMITER}rtl`),
+  ];
+
+  current.sort();
+
+  return current;
+}
+
+function groupBy(items, getGroupKey) {
+  const groups = [];
+
+  for (const item of items) {
+    const key = getGroupKey(item);
+
+    let group = groups.find((x) => x.key === key);
+
+    if (!group) {
+      group = { key, items: [] };
+
+      groups.push(group);
+    }
+
+    group.items.push(item);
+  }
+
+  return groups;
+}
+
+function* chunkArray(items, size) {
+  size = size <= 0 ? 1 : size;
+
+  for (let i = 0; i < items.length; i += size) {
+    yield items.slice(i, i + size);
+  }
+}
+
+function getInputConfigs() {
+  const jsTestInputFiles = [
+    'config-file.js',
+    'src/file.js',
+    'src/file.test.js',
+  ];
+
+  const tsTestInputFiles = [
+    'config-file.ts',
+    'src/file.ts',
+    'src/file.test.ts',
+    'declaration-file.d.ts',
+  ];
+
+  const configs = [];
+
+  for (const variation of getConfigVariations()) {
+    for (const testInputFile of jsTestInputFiles) {
+      configs.push({ variation, testInputFile });
+    }
+
+    if (variation.split(TAG_DELIMITER).includes('typescript')) {
+      for (const testInputFile of tsTestInputFiles) {
+        configs.push({ variation, testInputFile });
+      }
+    }
+  }
+
+  return configs;
+}
+
+function getNonConfiguredRules(eslintConfigArray, resolvedConfigs) {
```

Comment:
Doc: The function name is not descriptive enough, this needs a comment to explain that this returns all the rules that were not enabled by any config